models/glif_psc.h

```diff
@@ -0,0 +1,354 @@
+C_m        double - Capacitance of the membrane in pF.
+t_ref      double - Duration of refractory time in ms.
+V_reset    double - Reset potential of the membrane in mV (glif 1-3).
+a_spike    double - Threshold addition following spike in mV.
+b_spike    double - Spike-induced threshold time constant in 1/ms.
+a_reset    double - Voltage fraction coefficient following spike.
+b_reset    double - Voltage addition following spike in mV.
+asc_init   double vector - Initial values of after-spike currents in pA.
+k          double vector - After-spike current time constants in 1/ms
+                           (kj in Equation (3) in [1]).
+asc_amps   double vector - After-spike current amplitudes in pA
+                           (deltaIj in Equation (7) in [1]).
+r          double vector - Current fraction following spike coefficients
+                           (fj in Equation (7) in [1]).
+a_voltage  double - Adaptation index of threshold - A 'leak-conductance'
+                    for the voltage-dependent component of the threshold
+                    in 1/ms (av in Equation (4) in [1]).
+b_voltage  double - Voltage-induced threshold time constant - Inverse of which
+                    is the time constant of the voltage-dependent component
+                    of the threshold in 1/ms (bv in Equation (4) in [1]).
+tau_syn    double vector - Rise time constants of the synaptic alpha function
+                           in ms.
+glif_model string - glif model type:
+                    "lif"/"glif_lif"/"1" - GLIF model 1,
+                    "lif_r"/"glif_lif_r"/"2" - GLIF model 2,
+                    "lif_asc"/"glif_lif_asc"/"3" - GLIF model 3,
+                    "lif_r_asc"/"glif_lif_r_asc"/"4" - GLIF model 4, or
+                    "lif_r_asc_a"/"glif_lif_r_asc_a"/"5" - GLIF model 5.
+
+References:
+[1] Teeter C, Iyer R, Menon V, Gouwens N, Feng D, Berg J, Szafer A,
+    Cain N, Zeng H, Hawrylycz M, Koch C, & Mihalas S (2018)
+    Generalized leaky integrate-and-fire models classify multiple neuron types.
+    Nature Communications 9:709.
+
+Author: Binghuang Cai and Kael Dai @ Allen Institute for Brain Science
+*/
+
+namespace nest
+{
+
+class glif_psc : public nest::Archiving_Node
+{
+public:
+  glif_psc();
+
+  glif_psc( const glif_psc& );
+
+  using nest::Node::handle;
+  using nest::Node::handles_test_event;
+
+  nest::port send_test_event( nest::Node&, nest::port, nest::synindex, bool );
+
+  void handle( nest::SpikeEvent& );
+  void handle( nest::CurrentEvent& );
+  void handle( nest::DataLoggingRequest& );
+
+  nest::port handles_test_event( nest::SpikeEvent&, nest::port );
+  nest::port handles_test_event( nest::CurrentEvent&, nest::port );
+  nest::port handles_test_event( nest::DataLoggingRequest&, nest::port );
+
+  void get_status( DictionaryDatum& ) const;
+  void set_status( const DictionaryDatum& );
+
+  typedef std::string model_type;
+  std::map< std::string, long > model_type_lu = { { "lif", 1 },
```

Comment:
This should probably be revised, see comment above. If not, it should be `const` and private.
What is the meaning of `_lu` at the end of the name?